aspnetcore/performance/caching/memory.md

```diff
@@ -1,164 +1,111 @@
+[!code-html[Main](memory/sample/webcache/views/home/index.cshtml)]
 
-Possible `eviction reasons` are:
+The `DateTime.Now` value will stay in the cache while there are requests within the timeout period (and not enough memory pressure to evict the cached entry). The image below shows the cached value is older than the current value:
 
-**None**
+![Index view with two different times displayed](memory/_static/time.png)
 
-No reason known.
+The following code uses [GetOrCreate](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) and [GetOrCreateAsync](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) to cache data. 
 
-**Removed**
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet2&highlight=3,13-14)]
 
-The item was manually removed by a call to `Remove()`
+The following code calls [Get](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) to fetch the cached time:
 
-**Replaced**
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_gct)]
 
-The item was overwritten.
+See [IMemoryCache methods](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache) and [CacheExtensions methods](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheextensions) for a description of the cache methods.
 
-**Expired**
+## Using `MemoryCacheEntryOptions`
 
-The item timed out.
+The following sample:
 
-**TokenExpired**
+- Sets the absolute expiration time. This is the maximum time the entry can be cached and prevents the item from becoming too stale when the sliding expiration is continuously renewed.
+- Sets a sliding expiration time. Requests that access this cached item will reset the sliding expiration clock.
+- Sets the [cache priority](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.cacheitempriority) to `CacheItemPriority.NeverRemove`. 
+- Sets a [PostEvictionDelegate](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.postevictiondelegate) that will be called after the entry is evicted from the cache. The callback is run on a different thread from the code that removes the item from the cache.
 
-The token the item depended upon fired an event.
+[!code-csharp[Main](memory/sample/WebCache/Controllers/HomeController.cs?name=snippet_et)]
 
-**Capacity**
+## Cache dependencies
 
-The item was removed as part of the cache's memory management process.
+The following sample shows how to expire a cache entry if a dependent entry expires. A `CancellationTokenSource` is added to the cached item. When `Cancel` is called on the token, both cache entries are evicted. 
```

Comment:
A `CancellationChangeToken` (technically it only has to be a IChangeToken) was added not `CancellationTokenSource`.